cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -591,7 +591,7 @@ public Void call() throws Exception {
       });
     }
 
-    ApplicationSpecification appSpec = store.getApplication(appId);
+    ApplicationSpecification appSpec = store.getApplication(appId.toEntityId());
```

Comment:
should not do ``appId.toEntityId`` multiple times. It is already done on line 548 and 550. Better to extract to a variable.